crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -1656,20 +1659,64 @@ impl AuthorityPerEpochStore {
         };
         // This is stored as a vector<u8> in Move, so we double-deserialize to get back
         //`StoredExecutionTimeObservations`.
-        let Ok::<Vec<u8>, _>(stored_observations_bytes) = get_dynamic_field_from_store(
-            object_store,
-            system_state.extra_fields.id.id.bytes,
-            &EXTRA_FIELD_EXECUTION_TIME_ESTIMATES_KEY,
-        ) else {
-            warn!(
-                "Could not find stored execution time observations. This should only happen in the first epcoh where ExecutionTimeEstimate mode is enabled."
-            );
-            return itertools::Either::Left(std::iter::empty());
+        let stored_observations = if protocol_config.enable_observation_chunking() {
+            if let Ok::<u64, _>(chunk_count) = get_dynamic_field_from_store(
```

Comment:
on this line you might want to avoid falling back to the old version on errors other than `DynamicFieldReadError` (when it's not found)